pkg/restore/restore.go

```diff
@@ -711,6 +712,7 @@ func (ctx *context) restoreResource(resource, targetNamespace, originalNamespace
 func (ctx *context) getResourceClient(groupResource schema.GroupResource, obj *unstructured.Unstructured, namespace string) (client.Dynamic, error) {
 	key := resourceClientKey{
 		resource:  groupResource,
```

Comment:
if you make the above change, then resource can be set to `groupResource.WithVersion(obj.GroupVersionKind().Version)`